core/src/main/java/org/neo4j/ogm/metadata/ClassInfo.java

```diff
@@ -223,7 +224,18 @@ public ClassInfo directSuperclass() {
      * any, never <code>null</code>
      */
     public Collection<String> staticLabels() {
-        return collectLabels(new ArrayList<>());
+
+        Collection<String> knownStaticLabels = this.staticLabels;
+        if (knownStaticLabels == null) {
+            synchronized (this) {
+                knownStaticLabels = this.staticLabels;
+                if (knownStaticLabels == null) {
+                    this.staticLabels = Collections.unmodifiableCollection(collectLabels(new ArrayList<>()));
```

Comment:
Could we please get rid of the `new ArrayList` here and create and return it in the `collectLabels` function? This would need adjustments in the `collectLabels` but will feel much better IMHO.

Reply:
Was there before, but totally agree.